src/relay/backend/contrib/dnnl/query_layout.cc

```diff
@@ -0,0 +1,183 @@
+#include <tvm/relay/attrs/nn.h>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/transform.h>
+#include <tvm/relay/type.h>
+#include <tvm/runtime/module.h>
+#include <tvm/runtime/registry.h>
+
+#include <fstream>
+#include <numeric>
+#include <sstream>
+#include "dnnl.hpp"
+
+#include "../../utils.h"
+
+using dim_t = dnnl_dim_t;
+using dims_t = dnnl_dims_t;
+
+namespace tvm {
+namespace relay {
+namespace contrib {
+
+template <typename T, typename U>
+inline void array_set(T *arr, const U &val, size_t size) {
+    for (size_t i = 0; i < size; ++i)
+        arr[i] = static_cast<T>(val);
+}
+
+template <typename T>
+inline void array_copy(T *dst, const T *src, size_t size) {
+    for (size_t i = 0; i < size; ++i)
+        dst[i] = src[i];

```

Comment:
memset and memcpy would be optimal for elemenwise copy.

Reply:
memset and memcpy encounter the "overflow" warning.